sql/connect/client/jdbc/src/main/scala/org/apache/spark/sql/connect/client/jdbc/SparkConnectResultSet.scala

```diff
@@ -253,13 +263,25 @@ class SparkConnectResultSet(
   override def getBigDecimal(columnLabel: String): java.math.BigDecimal =
     throw new SQLFeatureNotSupportedException
 
-  override def isBeforeFirst: Boolean = throw new SQLFeatureNotSupportedException
+  override def isBeforeFirst: Boolean = {
+    checkOpen()
+    cursor < 1
```

Comment:
Oh wait, should `isBeforeFirst` return `false` if the result contains zero rows?
https://docs.oracle.com/en/java/javase/17/docs/api/java.sql/java/sql/ResultSet.html#isBeforeFirst()

Reply:
@sarutak thanks for catching this, fixed.